eth_defi/aave_v3/events.py

```diff
@@ -0,0 +1,311 @@
+    :param max_workers:
+        How many threads to allocate for JSON-RPC IO.
+        You can increase your EVM node output a bit by making a lot of parallel requests,
+        until you exhaust your nodes IO capacity. Experiement with different values
+        and see how your node performs.
+    :param log_info: Which function to use to output info messages about the progress
+    """
+    token_cache = TokenCache()
+    http_adapter = HTTPAdapter(pool_connections=max_workers, pool_maxsize=max_workers)
+    web3_factory = TunedWeb3Factory(json_rpc_url, http_adapter)
+    web3 = web3_factory(token_cache)
+    executor = create_thread_pool_executor(web3_factory, token_cache, max_workers=max_workers)
+    event_mapping = get_event_mapping(web3)
+    contract_events = [event_data["contract_event"] for event_data in event_mapping.values()]
+
+    # Start scanning
+    restored, restored_start_block = state.restore_state(start_block)
+    original_block_range = end_block - start_block
+
+    if restored:
+        log_info(f"Restored previous scan state, data until block {restored_start_block:,}, we are skipping {restored_start_block - start_block:,} blocks out of {original_block_range:,} total")
+    else:
+        log_info(
+            f"No previous scan done, starting fresh from block {start_block:,}, total {original_block_range:,} blocks",
+        )
+
+    # Prepare local buffers and files.
+    # Buffers is a context dictionary that is passed around
+    # by the event scanner.
+    buffers = {}
+
+    for event_name, mapping in event_mapping.items():
+        file_path = f"{output_folder}/aave-v3-{aave_network_name.lower()}-{event_name.lower()}.csv"
+        exists_already = Path(file_path).exists()
+        file_handler = open(file_path, "a")
+        csv_writer = csv.DictWriter(file_handler, fieldnames=mapping["field_names"])
+        if not exists_already:
+            csv_writer.writeheader()
+
+        # For each event, we have its own
+        # counters and handlers in the context dictionary
+        buffers[event_name] = {
+            "buffer": [],
+            "total": 0,
+            "file_handler": file_handler,
+            "csv_writer": csv_writer,
+        }
+
+    log_info(f"Scanning block range {restored_start_block:,} - {end_block:,}")
+    with tqdm(total=end_block - restored_start_block) as progress_bar:
+        #  1. update the progress bar
+        #  2. save any events in the buffer in to a file in one go
+        def update_progress(
+            current_block,
+            start_block,
+            end_block,
+            chunk_size: int,
+            total_events: int,
+            last_timestamp: int,
+            context: TokenCache,
+        ):
+            nonlocal buffers
+
+            if last_timestamp:
+                # Display progress with the date information
+                d = datetime.datetime.utcfromtimestamp(last_timestamp)
+                formatted_time = d.strftime("%d-%m-%Y")
```

Comment:
Is there a specific reason for not using the ISO format here? I think we already use it pretty consistently elsewhere across our system, and using a different format here could confuse someone.

Reply:
This is actually inherited by copy-paste from uniswap-v3/events.py. I can change it to ISO order in both.